wire-ios-sync-engine/Source/Synchronization/Strategies/UserClientRequestStrategy.swift

```diff
@@ -262,14 +285,46 @@ public final class UserClientRequestStrategy: ZMObjectSyncStrategy, ZMObjectStra
 
     public func request(forInserting managedObject: ZMManagedObject, forKeys keys: Set<String>?, apiVersion: APIVersion) -> ZMUpstreamRequest? {
         guard let client = managedObject as? UserClient else { fatal("Called requestForInsertingObject() on \(managedObject.safeForLoggingDescription)") }
+        guard let prekeys = clientRegistrationStatus?._prekeys else {
+            fatal("Asked to insert client when there's no prekeys available")
+        }
+        guard let lastResortPrekey = clientRegistrationStatus?._lastResortPrekey else {
+            fatal("Asked to insert client when there's no last resort prekey available")
+        }
+
         return try? requestsFactory.registerClientRequest(
                 client,
                 credentials: clientRegistrationStatus?.emailCredentials,
                 cookieLabel: CookieLabel.current.value,
+                prekeys: prekeys,
+                lastRestortPrekey: lastResortPrekey,
                 apiVersion: apiVersion
             )
     }
 
+    public func shouldCreateRequest(toSyncObject managedObject: ZMManagedObject, forKeys keys: Set<String>, withSync sync: Any) -> Bool {
+        if keys.contains(ZMUserClientNumberOfKeysRemainingKey), let userClient = managedObject as? UserClient {
+            if userClient.numberOfKeysRemaining >= minNumberOfRemainingKeys {
+                return false
+            } else if clientUpdateStatus?.currentPhase == .waitingForPrekeys {
+                clientUpdateStatus?.willGeneratePrekeys()
+                let groups = managedObjectContext?.enterAllGroupsExceptSecondary()
+                Task {
+                    let prekeys = try await prekeyGenerator.generatePrekeys()
+                    let lastResortPrekey = try await prekeyGenerator.generateLastResortPrekey()
```

Comment:
@typfel do we need to generate the lastresortPrekey here as we don't do anything with it ?

Reply:
No we don't. copy & paste error